### DISCUSSION TITLE
Appmetrica crash reporting on Exmol

### DISCUSSION THREAD

**Discussion by armanhossiny** [axmolengine/axmol#1700] [General]
I have a question that I need guidance and help with, and it's not directly related to the Exmol engine itself.

We have been working on a game for three years that was designed with cocos2dx v4. Three months ago, we migrated the game to Exmol (which we are very happy about, and we would like to thank all the guys who updated cocos2dx).

We use Appmetrica for crash reporting, but after the migration, the symbol files for uploading to Appmetrica are not generated correctly. I think the problem is with gradle incompatibility, and I've been working on this problem for several days but it hasn't been solved.

Is there anyone who can help? We have to release this week, but the game's crash reporting is not working (we can't use Firebase because it's not available for Iran). :(

**Comment by rh101** [comment]
> We use Appmetrica for crash reporting, but after the migration, the symbol files for uploading to Appmetrica are not generated correctly. I think the problem is with gradle incompatibility, and I've been working on this problem for several days but it hasn't been solved.

Can you give more details regarding this? What do you mean by "not generated correctly".  Are they being generated but corrupt in some way, or are they just not being generated at all?  Also, what makes you suspect that it is gradle incompatibility?  Which gradle version is being used?

**Comment by armanhossiny** [comment]
When we used cocos2dx and installed AppMetrica on it, a new folder named symbols was created in the build folder, which was zipped to the size of the game. However, in Axmol, this symbols folder is not created and cannot be uploaded to the AppMetrica website.

cocos2dx symbols:
![image](https://github.com/axmolengine/axmol/assets/42970068/0851531b-2b2a-4305-b2b1-46718ecb6d20)
![image](https://github.com/axmolengine/axmol/assets/42970068/10ab4bc3-9365-44f9-a6c1-98464149e675)

cocos2dx v4,  code Appmetrica Gradle 6.7.1
![image](https://github.com/axmolengine/axmol/assets/42970068/d7224b5c-fd86-4c87-a4b4-8c61c3f5d270)

Axmol 2.1.1, code Appmetrica Gradle 8.5:
(I rewrote the selected code in the photo in several other ways, but it didn't work.)
![image](https://github.com/axmolengine/axmol/assets/42970068/9d04dd62-e4ae-4d56-b759-06cf6802b921)

error:
![image](https://github.com/axmolengine/axmol/assets/42970068/5bf10992-6c72-40e9-871e-74514978e329)

link crash report AppMetrica:
(I think the documentation is not updated)
https://appmetrica.yandex.com/docs/en/sdk/android/analytics/android-build-plugin

last version AppMetrica
https://github.com/appmetrica/appmetrica-sdk-android

**Reply by rh101** [reply]
The project template has this section in it:
```
project.afterEvaluate {
    android.applicationVariants.configureEach { variant ->
        def variantName = variant.name.capitalize()
        def externalNativeBuild = tasks.named("externalNativeBuild${variantName}")
        if (externalNativeBuild) {
            def copyContentToAssets = tasks.named("copy${variantName}ContentToAssets")
            copyContentToAssets
            copyContentToAssets.get().dependsOn externalNativeBuild
            tasks.named("compile${variantName}JavaWithJavac").get().dependsOn copyContentToAssets
        }
    }
}
```

So, what happens if you add the relevant Appmetrica section to it like this:
```
project.afterEvaluate {
    android.applicationVariants.configureEach { variant ->
        def variantName = variant.name.capitalize()
        def externalNativeBuild = tasks.named("externalNativeBuild${variantName}")
        if (externalNativeBuild) {
            def copyContentToAssets = tasks.named("copy${variantName}ContentToAssets")
            copyContentToAssets
            copyContentToAssets.get().dependsOn externalNativeBuild
            tasks.named("compile${variantName}JavaWithJavac").get().dependsOn copyContentToAssets
        }

        // Appmetrica
        def uploadTaskName = "upload${variantName}Symbols"
        project.tasks.configureEach { task ->
            if (task.name == uploadTaskName) {
                variant.assembleProvider.configure { finalizedBy(task) } // If you use apk
                project.tasks.named("bundle${variantName}") { finalizedBy(task) } // If you use aab
            }
        }
    }
}
```

**Reply by armanhossiny** [reply]
Thanks, I had also tried this method but it gives an error and the project does not sync.(line 247 and 248).
![image](https://github.com/axmolengine/axmol/assets/42970068/3c0487af-98b7-4bfd-a928-9b665cec7763)

error:
![image](https://github.com/axmolengine/axmol/assets/42970068/35971eb0-4e7f-476b-9c09-43217339898a)

**Reply by rh101** [reply]
You should contact the developers of Appmetrica about this.

The only way I could see the uploadReleaseSymbols task was to set this to `true`:
`
    ndk {    // Optional.
        enable = { applicationVariant -> true }
    }
`

When that value is set to true, then comes up with the `DefaultTaskContainer#NamedDomainObjectProvider.configure(Action) on task set cannot be executed in the current context.` error.
